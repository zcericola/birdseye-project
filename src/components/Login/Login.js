/* eslint react/prop-types: 0 */
import React from 'react';
import useStyles from './loginStyles';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import logo from '../../assets/birdseye_logo.png';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';


const Login = ( props ) => {

    const signIn = () => () => {
        //sign in logic here
        //validate inputs and if valid, allow user to login
        return props.history.push( '/dashboard' );
    };
    const {
        loginContainer
        , formContainer
        , bgContainer
        , formFieldsContainer
        , formField
        , header
        , title
        , primaryBtn
        , secondaryBtn
        , label
        , textFieldWrapper
        , passwordLink
        , birdsEyeLogo
        , loginBtn
        , signInButton

    } = useStyles();
    return (
        <Grid container
            justify='center'
            alignItems='center'
            className={loginContainer}>
            <Grid
                item
                className={formContainer}
                xs={12} md={5}
            >
                <Grid item
                    xs={12}

                >
                    <div className={header}>
                        <img src={logo}
                            className={birdsEyeLogo}
                            alt='Birdseye logo'
                        />
                        <Typography
                            variant='h5'
                            className={title}
                        >
                            Welcome to Birdseye!
                        </Typography>
                    </div>
                </Grid>
                <Grid container
                    direction='column'
                    justify='center'
                    alignItems='center'
                    className={formFieldsContainer}
                >
                    <Grid item
                        className={textFieldWrapper}
                    >
                        <label className={label}>Email Address</label>
                        <TextField
                            className={formField}
                            fullWidth
                            variant='outlined'
                            margin='dense'
                        />
                    </Grid>
                    <Grid item
                        className={textFieldWrapper}
                    >
                        <label className={label}>Password</label>
                        <TextField
                            className={formField}
                            fullWidth
                            type='password'
                            variant='outlined'
                            margin='dense'
                        />
                    </Grid>
                    <Button
                        variant='outlined'
                        className={loginBtn}
                        margin='normal'
                        style={{ background: 'black', color: 'white' }}
                        onClick={signIn()}
                    >
                        Sign In
                    </Button>
                    <Button
                        variant='outlined'
                        className={loginBtn}
                        margin='normal'

                    >
                        Create An Account
                    </Button>
                    <Button
                        spacing={2}
                        component='span'
                        className={passwordLink}>
                        Forgot Password
                    </Button>



                </Grid>
            </Grid>
            <Grid
                item xs={12} md={7}
                className={bgContainer}
            >
            </Grid>


        </Grid >
    );

};

export default Login;