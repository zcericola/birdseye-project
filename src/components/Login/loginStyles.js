import { makeStyles } from '@material-ui/styles';
import loginBg from '../../assets/login_bg.png';

const useStyles = makeStyles({
    loginContainer: {
        height: '100%'
    }
    , formContainer: {
        height: '100vh'


    }
    , header: {

        height: '30vh'
        , width: '50%'
        , display: 'flex'
        , padding: 0
        , margin: 0
        , flexDirection: 'column'
        , justifyContent: 'space-evenly'
    }
    , title: {
        color: '#444'

    }
    , birdsEyeLogo: {
        height: 50
        , width: 60


    }
    , passwordLink: {
        fontSize: 10
        , color: '#777'
        , textTransform: 'uppercase'

    }
    , bgContainer: {
        height: '100vh'
        , backgroundImage: `url(${loginBg})`
        , backgroundSize: '100% 100%'


    }
    , formFieldsContainer: {


    }
    , textFieldWrapper: {
        width: '50%'

    }
    , formField: {
        [`& fieldset`]: {
            borderRadius: 0

        }

    }
    , label: {
        fontSize: 12
        , position: 'relative'
        , top: 5
    }

    , loginBtn: {
        width: '50%'
        , borderRadius: 0
        , margin: 10

    }

});

export default useStyles;