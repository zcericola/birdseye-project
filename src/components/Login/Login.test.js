import React from 'react';
import ReactDOM from 'react-dom';
import { mount, debug } from 'enzyme';
import Login from './Login';
import '../../setup';
import renderer from 'react-test-renderer';
import Typography from '@material-ui/core/Typography';

describe('Login Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree).toMatchSnapshot();

    });
    it('Login renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Login />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('Has a Login Container', () => {
        const wrapper = mount(<Login />);
        const container = wrapper.find('.loginContainer');
        expect(wrapper).toEqual({});
        wrapper.unmount();

    });
    it('Has the correct title', () => {
        const wrapper = mount(<Login />);
        const welcomeText = wrapper.find(Typography).text();
        expect(welcomeText).toEqual('Welcome to Birdseye!');
        wrapper.unmount();

    });

})
