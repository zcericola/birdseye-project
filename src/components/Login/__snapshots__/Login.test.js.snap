// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches the snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-loginContainer-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root makeStyles-formContainer-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="makeStyles-header-3"
      >
        <img
          alt="Birdseye logo"
          className="makeStyles-birdsEyeLogo-5"
          src="birdseye_logo.png"
        />
        <h5
          className="MuiTypography-root makeStyles-title-4 MuiTypography-h5"
        >
          Welcome to Birdseye!
        </h5>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-formFieldsContainer-8 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root makeStyles-textFieldWrapper-9 MuiGrid-item"
      >
        <label
          className="makeStyles-label-11"
        >
          Email Address
        </label>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-formField-10 MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-181 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-182"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-textFieldWrapper-9 MuiGrid-item"
      >
        <label
          className="makeStyles-label-11"
        >
          Password
        </label>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-formField-10 MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-181 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-182"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
            />
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-loginBtn-12 MuiButton-outlined"
        disabled={false}
        margin="normal"
        onBlur={[Function]}
        onClick={[Function]}
        onDragEnd={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "black",
            "color": "white",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sign In
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-loginBtn-12 MuiButton-outlined"
        disabled={false}
        margin="normal"
        onBlur={[Function]}
        onDragEnd={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Create An Account
        </span>
      </button>
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root makeStyles-passwordLink-6 MuiButton-text"
        onBlur={[Function]}
        onDragEnd={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        spacing={2}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Forgot Password
        </span>
      </span>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-bgContainer-7 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
  />
</div>
`;
